test/test_dynamics.jl

```diff
@@ -392,6 +396,77 @@ function inertial_oscillations_work_with_rotation_in_different_axis(arch, FT)
     return nothing
 end
 
+# for `test_velocity_combination`
+struct SinkingParticles <: AbstractBiogeochemistry end
+biogeochemical_drift_velocity(::SinkingParticles, ::Val{:A}) = (u = ConstantField(-1), v = ZeroField(), w = ZeroField())
+
+function test_velocity_combination(arch)
+    grid = RectilinearGrid(arch; size = 40, halo = 5, x = (0, 1), topology = (Periodic, Flat, Flat))
+
+    u(args...) = -1
+
+    model = NonhydrostaticModel(; grid, 
+                                  tracers=:A, 
+                                  advection = UpwindBiased(),
+                                  background_fields = (; u))
+    
+    A₀(x, y, z) = x > 0.4 && x < 0.6 ? 1.0 : 0.0
+
+    set!(model, u = 1, A = A₀)
+
+    Δt = 0.2 / 1.0 * grid.Δxᶜᵃᵃ[1]
```

Comment:
Use functions to access grid metrics, don't write code that explicitly references grid properties (it doesn't generalize and will break if the names change)